tests/test_notation_sets.py

```
@@ -0,0 +1,182 @@
+    assert m_g == "M:B:C:g"
+
+    comp = canonical_ir_encode("composition", m1=m_f, m2=m_g)
+    assert comp == f"COMP({m_f},{m_g})"
+
+    id_a = canonical_ir_encode("identity", obj="A")
+    assert id_a == "ID(A)"
+
+    func = canonical_ir_encode("functor", src_cat="C", tgt_cat="D", label="F")
+    assert func == "F:C:D:F"
+
+    nat_trans = canonical_ir_encode("natural_transformation", src_fun="F", tgt_fun="G", label="alpha")
```

Pylint (4.0.7)

[convention] Line too long (102/100)

(C0301)